cmd/olm/main.go

```diff
@@ -106,6 +127,65 @@ func main() {
 	http.Handle("/metrics", prometheus.Handler())
 	go http.ListenAndServe(":8080", nil)
 
-	_, done := operator.Run(stopCh)
+	ready, done := operator.Run(stopCh)
+	<-ready
+
+	if *writeStatusName != "" {
+		_, err := opClient.GetAPIService(openshiftConfigServiceName)
+		if k8serrors.IsNotFound(err) {
+			log.Info("Did not find openshift config API service, skipping status update")
+		} else if err != nil {
+			log.Fatalf("APIService check error: %v", err)
+		} else {
+			existing, err := configClient.ClusterOperators().Get(*writeStatusName, metav1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				log.Info("Existing cluster operator not found, creating")
+				created, err := configClient.ClusterOperators().Create(&configv1.ClusterOperator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: *writeStatusName,
+					},
+				})
+				if err != nil {
+					log.Fatalf("ClusterOperator create failed: %v\n", err)
+				}
+
+				created.Status = configv1.ClusterOperatorStatus{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						configv1.ClusterOperatorStatusCondition{
+							Type:               configv1.OperatorAvailable,
+							Status:             configv1.ConditionTrue,
+							Message:            fmt.Sprintf("Done deploying %s.", olmversion.OLMVersion),
+							LastTransitionTime: metav1.Now(),
+						},
+					},
+					Version: olmversion.Full(),
+				}
+				_, err = configClient.ClusterOperators().UpdateStatus(created)
+				if err != nil {
+					log.Fatalf("ClusterOperator update status failed: %v", err)
+				}
+			} else if err != nil {
+				log.Fatalf("ClusterOperators get failed: %v", err)
+			} else {
+				resourcemerge.SetOperatorStatusCondition(&existing.Status.Conditions, configv1.ClusterOperatorStatusCondition{
```

Comment:
You should be using [this](https://github.com/openshift/library-go/tree/master/pkg/config/clusteroperator/v1helpers) package